src/containers/Review/ReviewHome.tsx

```diff
@@ -55,10 +66,96 @@ const ReviewHome: React.FC<ReviewHomeProps> = ({ assignments, structure }) => {
                 }}
               />
             ))}
-        </Segment>
-      ))}
+          </Segment>
+        ))}
     </>
   )
 }
 
+type ReviewerAndStageSelectionProps = {
+  filters: Filters | null
+  setFilters: (filters: Filters) => void
+  structure: FullStructure
+  assignments: AssignmentDetailsNEW[]
+}
+
+const ReviewerAndStageSelection: React.FC<ReviewerAndStageSelectionProps> = ({
+  assignments,
+  structure,
+  filters,
+  setFilters,
+}) => {
+  const {
+    userState: { currentUser },
+  } = useUserState()
+
+  useEffect(() => {
+    setFilters({
+      selectedReviewer: currentUser?.userId as number,
+      selectedStage: structure.info.current?.stage.id as number,
+    })
+  }, [])
+
+  const changeFilters = (filterType: keyof Filters) => (_: any, { value }: any) => {
+    if (filters) setFilters({ ...filters, [filterType]: value })
+  }
+
+  if (!filters) return null
+
+  return (
+    <Grid columns="equal">
+      <Grid.Column floated="left">
+        Show tasks for{' '}
+        <Dropdown
+          options={getReviewerOptions(assignments, currentUser?.userId as number)}
+          value={filters?.selectedReviewer}
+          onChange={changeFilters('selectedReviewer')}
+        />
+      </Grid.Column>
+      <Grid.Column floated="right" textAlign="right">
+        STAGE{' '}
+        <Dropdown
+          options={getStageOptions(structure, assignments)}
+          value={filters?.selectedStage}
+          onChange={changeFilters('selectedStage')}
+        />
+      </Grid.Column>
+    </Grid>
+  )
+}
+
+const getStageOptions = (structure: FullStructure, assignments: AssignmentDetailsNEW[]) =>
+  structure.stages
+    .filter(({ id }) => assignments.some(({ stage }) => id === stage.id))
+    .map(({ id, title }) => ({
+      key: id,
+      value: id,
+      text: title,
+    }))
+
+const getReviewerOptions = (assignments: AssignmentDetailsNEW[], currentUserId: number) => {
+  const reviewerOptions: { value: number; key: number; text: string }[] = [
+    {
+      value: ALL_REVIEWERS,
+      key: ALL_REVIEWERS,
+      text: 'All',
+    },
+    {
+      value: currentUserId,
+      key: currentUserId,
+      text: 'Yourself',
+    },
+  ]
+  assignments.forEach(({ reviewer: { id, firstName, lastName } }) => {
+    if (id === currentUserId || !id || !firstName || !lastName) return
+    reviewerOptions.push({
+      value: id,
+      key: id,
+      text: `${firstName} ${lastName}`,
+    })
+  })
+
+  return reviewerOptions
+}
```

Comment:
I think those 2 should be moved to some helper function - for better readability of this. The name of the function already tells the story

Reply:
sorry, missed it: 92a4c99

Reply:
Hmm, I was also thinking it's getting a bit much for ReviewHome page. What do you think, if i move the whole component `ReviewerAndStageSelection` along with it's helper methods, would that be ok ?

Reply:
Yes, that will be nice